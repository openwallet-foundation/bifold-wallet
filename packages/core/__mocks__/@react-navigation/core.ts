const navigate = jest.fn()
const dispatch = jest.fn()
const replace = jest.fn()
const navigation = {
  __timestamp: process.hrtime(),
  navigate,
  replace,
  setOptions: jest.fn(),
  getParent: () => {
    return {
      navigate,
      dispatch,
      replace,
    }
  },
  getState: jest.fn(() => ({
    index: jest.fn(),
  })),
  goBack: jest.fn(),
  pop: jest.fn(),
  reset: jest.fn(),
  isFocused: () => true,
  dispatch,
}

const useNavigation = () => {
  return navigation
}

const useIsFocused = () => {
  return true
}

const CommonActions = {
  navigate: jest.fn(),
  reset: jest.fn(),
  goBack: jest.fn(),
}

export { useNavigation, useIsFocused, CommonActions }
