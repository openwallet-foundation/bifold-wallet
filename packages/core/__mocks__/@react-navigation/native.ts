const navigate = jest.fn()
const dispatch = jest.fn()
const replace = jest.fn()
const navigation = {
  __timestamp: process.hrtime(),
  navigate,
  replace,
  setOptions: jest.fn(),
  getParent: () => {
    return {
      navigate,
      dispatch,
      replace,
    }
  },
  getState: jest.fn(() => ({
    index: jest.fn(),
  })),
  goBack: jest.fn(),
  pop: jest.fn(),
  reset: jest.fn(),
  isFocused: () => true,
  dispatch,
}

const useNavigation = () => {
  return navigation
}

const useIsFocused = () => {
  return true
}

const useRoute = jest.fn().mockReturnValue({ params: {} })

const CommonActions = {
  navigate: jest.fn(),
  reset: jest.fn(),
  goBack: jest.fn(),
}

const useFocusEffect = jest.fn()
const createNavigatorFactory = jest.fn()

export { useNavigation, useIsFocused, useRoute, useFocusEffect, createNavigatorFactory, CommonActions }
