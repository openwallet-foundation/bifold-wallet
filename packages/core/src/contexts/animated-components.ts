import { createContext, useContext } from 'react'

import { AnimatedComponents, animatedComponents } from '../animated-components'

export const AnimatedComponentsContext = createContext<AnimatedComponents>(animatedComponents)

export const AnimatedComponentsProvider = AnimatedComponentsContext.Provider

export const useAnimatedComponents = () => useContext(AnimatedComponentsContext)
