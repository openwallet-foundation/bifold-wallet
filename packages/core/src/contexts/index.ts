import * as reducers from './reducers'
import * as store from './store'
export { reducers, store }
