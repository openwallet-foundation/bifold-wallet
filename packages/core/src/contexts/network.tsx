import { NetInfoStateType, useNetInfo } from '@react-native-community/netinfo'
import { createContext, useContext, useState, useCallback, PropsWithChildren } from 'react'

import NetInfoModal from '../components/modals/NetInfoModal'

export interface NetworkContext {
  silentAssertConnectedNetwork: () => boolean | null
  assertNetworkConnected: () => boolean
  displayNetInfoModal: () => void
  hideNetInfoModal: () => void
  assertInternetReachable: () => boolean | null
}

export const NetworkContext = createContext<NetworkContext>(null as unknown as NetworkContext)

// NOTE: @react-native-community/netinfo can be configured to use whichever reachability check desired
// eg. isInternetReachable can be set to check a specific URL (like your mediator). See the docs here for more info:
// https://github.com/react-native-netinfo/react-native-netinfo?tab=readme-ov-file#configure
export const NetworkProvider = ({ children }: PropsWithChildren) => {
  const { isConnected, type, isInternetReachable } = useNetInfo()
  const [isNetInfoModalDisplayed, setIsNetInfoModalDisplayed] = useState<boolean>(false)

  const displayNetInfoModal = useCallback(() => {
    setIsNetInfoModalDisplayed(true)
  }, [])

  const hideNetInfoModal = useCallback(() => {
    setIsNetInfoModalDisplayed(false)
  }, [])

  /**
   * Returns null until the network state is known, then returns boolean
   * Useful for cases where we do not want to take action until the network state is known
   *
   * @returns {boolean | null} - `true` if the network is connected, `false` if not connected,
   *                             and `null` if the network status not yet known
   */
  const silentAssertConnectedNetwork = useCallback((): boolean | null => {
    return type === NetInfoStateType.unknown
      ? null
      : isConnected || [NetInfoStateType.wifi, NetInfoStateType.cellular].includes(type)
  }, [isConnected, type])

  /**
   * Strictly asserts that the network is connected. Will return false even if
   * the network state is not yet known - in this case it will also display the
   * NetInfoModal
   * Useful for cases where we must be sure of connectivity before proceeding
   *
   * @returns {boolean} - `true` if the network is checked and connected, otherwise `false`
   */
  const assertNetworkConnected = useCallback(() => {
    const connectionConfirmed = silentAssertConnectedNetwork() === true
    if (!connectionConfirmed) {
      displayNetInfoModal()
    }

    return connectionConfirmed
  }, [silentAssertConnectedNetwork, displayNetInfoModal])

  const assertInternetReachable = useCallback((): boolean | null => {
    return isInternetReachable
  }, [isInternetReachable])

  return (
    <NetworkContext.Provider
      value={{
        silentAssertConnectedNetwork,
        assertNetworkConnected,
        displayNetInfoModal,
        hideNetInfoModal,
        assertInternetReachable,
      }}
    >
      {children}
      <NetInfoModal visible={isNetInfoModalDisplayed} onSubmit={() => hideNetInfoModal()} />
    </NetworkContext.Provider>
  )
}

export const useNetwork = () => useContext(NetworkContext)
