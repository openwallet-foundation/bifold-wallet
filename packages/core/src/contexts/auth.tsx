// Dont remove the following import line or the pin check will fail when opening askar waller
import '@hyperledger/aries-askar-react-native'

import 'reflect-metadata'

import { AskarWallet } from '@credo-ts/askar'
import { ConsoleLogger, LogLevel, SigningProviderRegistry } from '@credo-ts/core'
import { useAgent } from '@credo-ts/react-hooks'
import { agentDependencies } from '@credo-ts/react-native'
import React, { createContext, useCallback, useContext, useState } from 'react'
import { useTranslation } from 'react-i18next'

import { DispatchAction } from './reducers/store'
import { useStore } from './store'
import {
  isBiometricsActive,
  loadWalletSalt,
  loadWalletSecret,
  secretForPIN,
  storeWalletSecret,
  wipeWalletKey,
} from '../services/keychain'
import { WalletSecret } from '../types/security'
import { hashPIN } from '../utils/crypto'
import { migrateToAskar } from '../utils/migration'

export interface AuthContext {
  checkWalletPIN: (PIN: string) => Promise<boolean>
  getWalletSecret: () => Promise<WalletSecret | undefined>
  walletSecret?: WalletSecret
  removeSavedWalletSecret: () => void
  disableBiometrics: () => Promise<void>
  setPIN: (PIN: string) => Promise<void>
  commitWalletToKeychain: (useBiometry: boolean) => Promise<boolean>
  isBiometricsActive: () => Promise<boolean>
  rekeyWallet: (oldPin: string, newPin: string, useBiometry?: boolean) => Promise<boolean>
}

export const AuthContext = createContext<AuthContext>(null as unknown as AuthContext)

export const AuthProvider: React.FC<React.PropsWithChildren> = ({ children }) => {
  const [walletSecret, setWalletSecret] = useState<WalletSecret>()
  const [store, dispatch] = useStore()
  const { agent } = useAgent()
  const { t } = useTranslation()

  const setPIN = useCallback(async (PIN: string): Promise<void> => {
    const secret = await secretForPIN(PIN)
    await storeWalletSecret(secret)
  }, [])

  const getWalletSecret = useCallback(async (): Promise<WalletSecret | undefined> => {
    if (walletSecret) {
      return walletSecret
    }

    const secret = await loadWalletSecret(t('Biometry.UnlockPromptTitle'), t('Biometry.UnlockPromptDescription'))

    setWalletSecret(secret)

    return secret
  }, [t, walletSecret])

  const commitWalletToKeychain = useCallback(
    async (useBiometry: boolean): Promise<boolean> => {
      const secret = await getWalletSecret()
      if (!secret) {
        return false
      }

      // set did authenticate to true if we can get wallet credentials
      dispatch({
        type: DispatchAction.DID_AUTHENTICATE,
      })
      if (useBiometry) {
        await storeWalletSecret(secret, useBiometry)
      } else {
        // erase wallet key if biometrics is disabled
        await wipeWalletKey(useBiometry)
      }

      return true
    },
    [dispatch, getWalletSecret]
  )

  const checkWalletPIN = useCallback(
    async (PIN: string): Promise<boolean> => {
      try {
        const secret = await loadWalletSalt()

        if (!secret?.salt) {
          return false
        }

        const hash = await hashPIN(PIN, secret.salt)

        if (!store.migration.didMigrateToAskar) {
          await migrateToAskar(secret.id, hash)
          dispatch({
            type: DispatchAction.DID_MIGRATE_TO_ASKAR,
          })
        }

        // NOTE: We create an instance of AskarWallet, which is the underlying wallet that powers the app
        // we then open that instance with the provided id and key to verify their integrity
        const askarWallet = new AskarWallet(
          new ConsoleLogger(LogLevel.off),
          new agentDependencies.FileSystem(),
          new SigningProviderRegistry([])
        )
        await askarWallet.open({
          id: secret.id,
          key: hash,
        })

        await askarWallet.close()

        const fullSecret = await secretForPIN(PIN, secret.salt)
        setWalletSecret(fullSecret)
        return true
      } catch (e) {
        return false
      }
    },
    [dispatch, store.migration.didMigrateToAskar]
  )

  const removeSavedWalletSecret = useCallback(() => {
    setWalletSecret(undefined)
  }, [])

  const disableBiometrics = useCallback(async () => {
    await wipeWalletKey(true)
  }, [])

  const rekeyWallet = useCallback(
    async (oldPin: string, newPin: string, useBiometry?: boolean): Promise<boolean> => {
      try {
        if (!agent) {
          // no agent set, cannot rekey the wallet
          return false
        }
        // argon2.hash can sometimes generate an error
        const secret = await loadWalletSalt()
        if (!secret) {
          return false
        }

        const oldHash = await hashPIN(oldPin, secret.salt)
        const newSecret = await secretForPIN(newPin)
        const newHash = await hashPIN(newPin, newSecret.salt)
        if (!newSecret.key) {
          return false
        }

        await agent.wallet.close()
        // wallet.rotateKey calls open under the hood
        await agent.wallet.rotateKey({ id: secret.id, key: oldHash, rekey: newHash })

        await storeWalletSecret(newSecret, useBiometry)
        setWalletSecret(newSecret)
      } catch {
        return false
      }
      return true
    },
    [agent]
  )

  return (
    <AuthContext.Provider
      value={{
        checkWalletPIN,
        getWalletSecret,
        removeSavedWalletSecret,
        disableBiometrics,
        commitWalletToKeychain,
        setPIN,
        isBiometricsActive,
        rekeyWallet,
        walletSecret,
      }}
    >
      {children}
    </AuthContext.Provider>
  )
}

export const useAuth = () => useContext(AuthContext)
