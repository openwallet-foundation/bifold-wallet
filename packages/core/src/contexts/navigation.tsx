import { NavigationContainer, NavigationContainerRef } from '@react-navigation/native'
import React from 'react'
import { useTheme } from './theme'

export interface NavContainerProps extends React.PropsWithChildren {
  navigationRef: React.RefObject<NavigationContainerRef<any>> | null
}

const NavContainer = ({ navigationRef, children }: NavContainerProps) => {
  const { NavigationTheme } = useTheme()

  return (
    <NavigationContainer ref={navigationRef} theme={NavigationTheme}>
      {children}
    </NavigationContainer>
  )
}

export default NavContainer
