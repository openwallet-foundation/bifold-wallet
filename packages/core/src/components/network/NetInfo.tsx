import { useEffect, useCallback, useState } from 'react'
import { useTranslation } from 'react-i18next'
import Toast from 'react-native-toast-message'
import { useNetwork } from '../../contexts/network'
import { ToastType } from '../toast/BaseToast'

const NetInfo: React.FC = () => {
  const { assertInternetReachable } = useNetwork()
  const { t } = useTranslation()
  const [hasShown, setHasShown] = useState(false)

  const showNetworkWarning = useCallback(() => {
    setHasShown(true)
    Toast.show({
      type: ToastType.Error,
      autoHide: true,
      text1: t('NetInfo.NoInternetConnectionTitle'),
    })
  }, [t])

  useEffect(() => {
    const internetReachable = assertInternetReachable()
    if (internetReachable) {
      Toast.hide()
    }

    // Strict check for false, null means the network state is not yet known
    if (internetReachable === false && !hasShown) {
      showNetworkWarning()
    }
  }, [showNetworkWarning, assertInternetReachable, hasShown])

  return null
}

export default NetInfo
