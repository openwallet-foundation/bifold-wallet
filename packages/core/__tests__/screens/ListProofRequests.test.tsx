import { getProofRequestTemplates } from '@hyperledger/aries-bifold-verifier'
import mockRNCNetInfo from '@react-native-community/netinfo/jest/netinfo-mock'
import { act, fireEvent, render } from '@testing-library/react-native'
import React from 'react'

import { useNavigation as testUseNavigation } from '../../__mocks__/@react-navigation/native'
import { NetworkProvider } from '../../src/contexts/network'
import ListProofRequests from '../../src/screens/ListProofRequests'
import { BasicAppContext } from '../helpers/app'

jest.mock('react-native/Libraries/EventEmitter/NativeEventEmitter')
jest.mock('@react-native-community/netinfo', () => mockRNCNetInfo)
// eslint-disable-next-line @typescript-eslint/no-empty-function
jest.mock('react-native-localize', () => {})

jest.useFakeTimers({ legacyFakeTimers: true })
jest.spyOn(global, 'setTimeout')

const navigation = testUseNavigation()

describe('ListProofRequests Component', () => {
  const renderView = (params?: any) => {
    return render(
      <BasicAppContext>
        <NetworkProvider>
          <ListProofRequests navigation={navigation as any} route={{ params: params || {} } as any} />
        </NetworkProvider>
      </BasicAppContext>
    )
  }

  test('Renders correctly', async () => {
    // const tree = renderView()
    const tree = render(
      <BasicAppContext>
        <ListProofRequests navigation={navigation as any} route={{ params: {} } as any} />
      </BasicAppContext>
    )
    await act(async () => {})
    expect(tree).toMatchSnapshot()
  })

  test('Template names are human readable', async () => {
    const tree = renderView()

    const fullName = await tree.findByText('Student full name', { exact: true })
    const fullNameAndExpirationDate = await tree.findByText('Student full name and expiration date', { exact: false })

    expect(fullName).not.toBe(null)
    expect(fullNameAndExpirationDate).not.toBe(null)
  })

  test('Pressing on a request template takes the user to a proof request template detail screen', async () => {
    const tree = renderView()

    await act(async () => {
      const templateItemInstance = await tree.findByText('Student full name', { exact: true })

      fireEvent(templateItemInstance, 'press')

      expect(navigation.navigate).toBeCalledWith('Proof Request Details', {
        templateId: getProofRequestTemplates(false)[0].id,
      })
    })
  })
})
