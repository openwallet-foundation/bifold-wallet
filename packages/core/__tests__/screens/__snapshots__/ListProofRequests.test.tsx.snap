// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListProofRequests Component Renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    Object {
      "bottom": "off",
      "left": "additive",
      "right": "additive",
      "top": "off",
    }
  }
  style={
    Object {
      "elevation": 5,
      "flexGrow": 1,
      "margin": 24,
    }
  }
>
  <RCTScrollView
    ListEmptyComponent={[Function]}
    data={
      Array [
        Object {
          "description": "Verify the full name of a student",
          "id": "Aries:5:StudentFullName:0.0.1:indy",
          "name": "Student full name",
          "payload": Object {
            "data": Array [
              Object {
                "requestedAttributes": Array [
                  Object {
                    "name": "student_first_name",
                    "restrictions": Array [
                      Object {
                        "cred_def_id": "XUxBrVSALWHLeycAUhrNr9:3:CL:26293:student_card",
                      },
                    ],
                  },
                  Object {
                    "name": "student_last_name",
                    "restrictions": Array [
                      Object {
                        "cred_def_id": "XUxBrVSALWHLeycAUhrNr9:3:CL:26293:student_card",
                      },
                    ],
                  },
                ],
                "schema": "XUxBrVSALWHLeycAUhrNr9:3:CL:26293:Student Card",
              },
            ],
            "type": "anoncreds",
          },
          "version": "0.0.1",
        },
        Object {
          "description": "Verify that full name of a student and that he/she has a not expired student card.",
          "id": "Aries:5:StudentFullNameAndExpirationDate:0.0.1:indy",
          "name": "Student full name and expiration date",
          "payload": Object {
            "data": Array [
              Object {
                "requestedAttributes": Array [
                  Object {
                    "names": Array [
                      "student_first_name",
                      "student_last_name",
                    ],
                    "restrictions": Array [
                      Object {
                        "cred_def_id": "XUxBrVSALWHLeycAUhrNr9:3:CL:26293:student_card",
                      },
                    ],
                  },
                ],
                "requestedPredicates": Array [
                  Object {
                    "name": "expiry_date",
                    "predicateType": ">=",
                    "predicateValue": 20240101,
                    "restrictions": Array [
                      Object {
                        "cred_def_id": "XUxBrVSALWHLeycAUhrNr9:3:CL:26293:student_card",
                      },
                    ],
                  },
                ],
                "schema": "XUxBrVSALWHLeycAUhrNr9:3:CL:26293:Student Card",
              },
            ],
            "type": "anoncreds",
          },
          "version": "0.0.1",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "#000000",
      }
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityLabel="Screens.ProofRequestDetails"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            Object {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 8,
              "flexDirection": "row",
              "marginBottom": 10,
              "opacity": 1,
              "padding": 12,
            }
          }
          testID="com.ariesbifold:id/ProofRequestsCard"
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              maxFontSizeMultiplier={2}
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "normal",
                  },
                  Object {
                    "color": "#000000",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginBottom": 4,
                  },
                ]
              }
            >
              Student full name
            </Text>
            <Text
              maxFontSizeMultiplier={2}
              numberOfLines={2}
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "normal",
                  },
                  Object {
                    "color": "#000000",
                    "fontSize": 16,
                    "fontWeight": "normal",
                    "marginBottom": 4,
                  },
                ]
              }
            >
              Verify the full name of a student
            </Text>
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Object {
                    "color": "#000000",
                    "fontSize": 36,
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              î—Œ
            </Text>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityLabel="Screens.ProofRequestDetails"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            Object {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 8,
              "flexDirection": "row",
              "marginBottom": 10,
              "opacity": 1,
              "padding": 12,
            }
          }
          testID="com.ariesbifold:id/ProofRequestsCard"
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              maxFontSizeMultiplier={2}
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "normal",
                  },
                  Object {
                    "color": "#000000",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginBottom": 4,
                  },
                ]
              }
            >
              Student full name and expiration date
            </Text>
            <Text
              maxFontSizeMultiplier={2}
              numberOfLines={2}
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "normal",
                  },
                  Object {
                    "color": "#000000",
                    "fontSize": 16,
                    "fontWeight": "normal",
                    "marginBottom": 4,
                  },
                ]
              }
            >
              Verify that full name of a student and that he/she has a not expired student card.
            </Text>
            <Text
              maxFontSizeMultiplier={2}
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "normal",
                  },
                  Object {
                    "color": "#606060",
                    "fontSize": 12,
                  },
                ]
              }
            >
              Verifier.ZeroKnowledgeProof
            </Text>
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Object {
                    "color": "#000000",
                    "fontSize": 36,
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              î—Œ
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
